src/network/clientpackethandler.cpp

```diff
@@ -530,11 +531,15 @@ void Client::handleCommand_Movement(NetworkPacket* pkt)
 void Client::handleCommand_Fov(NetworkPacket *pkt)
 {
 	f32 fov;
-	bool is_multiplier;
-	*pkt >> fov >> is_multiplier;
+	bool is_multiplier = false;
+	f32 transition_time = 0.0f;
+
+	*pkt >> fov >> is_multiplier >> transition_time;
```

Comment:
you should test protocol version & bump it, else a client connecting to a old server will just crash its game session

Reply:
This isn't backwards-compatible?

Reply:
you will try to read something non existent on packet => failure

Reply:
failure == exception? If so, can I just have a try-catch with an empty catch block, like how itemdef packets are parsed?

Reply:
yep this iwll trigger an exception, you can find how to handle it on some other packet handlers

Reply:
you should just check whether there is more left in the stream after reading the fov, relying on an empty catch is probably not good for performance

Reply:
Thanks for your input. I just timed these operations, and both ways take around 0.001ms to complete (although the empty catch block occasionally takes up to 0.002ms) - I wouldn't be too worried about performance here.

And while checking for remaining data in the packet sounds like a better idea, there's the argument of consistency for the empty-catch block approach. I'll leave this as it is - if a core dev asks me to use the other way, I'll be happy to do so.